Assets/Scripts/Tags/Hazard.cs

```diff
@@ -1,45 +1,41 @@
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public class Hazard : MonoBehaviour {
+    [Tooltip("Defines how high and low this object will bob up and down, measured in world-space units.")]
+    [SerializeField] private float amplitude = 0.2f;
 
-    [SerializeField] private float verticalSpeed;
-    [SerializeField] private float amplitude;
+    [Tooltip("The rate at which this object will rotate over time, measured in degrees per second.")]
+    [FormerlySerializedAs("rotSpeed")]
+    [SerializeField] private float angularSpeed = 60;
 
-    [SerializeField] private float rotSpeed;
+    [Tooltip("Defines the range within which a vertical speed will be randomly chosen for this object.\n\n" +
+        "This object's vertical speed describes the frequency of the up and down bobbing motion.")]
+    [SerializeField] private Vector2 verticalSpeedRange = new Vector2(2, 3);
 
-    private Vector3 origPos;
-    private Vector3 tempPos;
-
-    private Vector3 rot = new Vector3(45f, 45f, 45f);
-
-    private float rotDegreesPerSecond = 45f;
-
-    private static System.Random rndFloat = new System.Random();
-    private static System.Random rndRot = new System.Random();
-
-    private void Start () {
-        tempPos = origPos = new Vector3(transform.position.x, (float) rndFloat.NextDouble() + .3f, transform.position.z);
-        verticalSpeed = (float)rndFloat.NextDouble();
-
-        //rot = Quaternion.Euler((float)rndRot.NextDouble(), (float)rndRot.NextDouble(), (float)rndRot.NextDouble());
+    private Vector3 originalPosition;
+    private float verticalSpeed;
+    private Vector3 rotationalAxis;
 
+    private void Start() {
+        originalPosition = transform.position;
+        verticalSpeed = Random.Range(verticalSpeedRange.x, verticalSpeedRange.y);
+        rotationalAxis = Random.onUnitSphere;
     }
 
-    private void FixedUpdate() {
-        RandomFloating();
-        Rotation();
+    private void Update() {
+        UpdatePosition();
+        UpdateRotation();
     }
 
-    private void RandomFloating() {
-        tempPos = origPos;
-        tempPos.y += Mathf.Sin(Time.realtimeSinceStartup * (verticalSpeed + 2)) * amplitude;
-        transform.position = tempPos;
+    private void UpdatePosition() {
+        Vector3 pos = originalPosition;
+        pos.y = originalPosition.y + amplitude * Mathf.Sin(Time.time * verticalSpeed);
+        transform.position = pos;
     }
 
-    private void Rotation() {
-        //float yAngle = transform.rotation.eulerAngles.y;
-
-        transform.rotation = Quaternion.Euler(rot * Time.deltaTime) * transform.rotation;
-        //transform.rotation = Quaternion.AngleAxis(yAngle + (Time.deltaTime * rotDegreesPerSecond), Vector3.up);
+    private void UpdateRotation() {
+        Quaternion deltaRotation = Quaternion.AngleAxis(angularSpeed * Time.deltaTime, rotationalAxis);
+        transform.rotation = deltaRotation * transform.rotation;
```

Comment:
You were VERY close!

I just deleted what was written here, and did the constant-speed rotation in the simplest way I could haha.
Hopefully this makes sense!
We can draw diagrams, if needed, to help explain what this all mathematically looks like.

Reply:
Okay! I would like to go over everything sometime this weekend if you have time :) Thank you so so SO very much!

You're the best :heart: